src/mixins/name-request-mixin.ts

```diff
@@ -37,7 +42,14 @@ export default class NameRequestMixin extends Mixins(DateMixin) {
 
     // ensure NR is valid
     const isNrValid = this.isNrValid(nrResponse)
-    if (!nrResponse || !isNrValid) {
+    const isNewNr = nrType === CorrectionTypes.CORRECT_NEW_NR
```

Comment:
OK, I'm pretty confused here. What is this variable exactly? Is it ever false? How does it affect the logic below -- isn't it always a new NR that we're validating?

I'm also confused about the scenario when either of the 2 following blocks is called.

Please add comments or review the variable names or review the scenarios.